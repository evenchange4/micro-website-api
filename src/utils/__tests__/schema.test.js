// @flow
const Joi = require('joi');
const schema = require('../schema');

it('should return result', () => {
  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
        },
      },
      schema,
    ),
  ).toMatchSnapshot();

  expect(
    Joi.validate(
      {
        query: {
          url: '123',
          selector: 'selector',
        },
      },
      schema,
    ),
  ).toMatchSnapshot('invalid uri format');

  expect(
    Joi.validate(
      {
        query: {},
      },
      schema,
    ),
  ).toMatchSnapshot('required');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: true,
        },
      },
      schema,
    ),
  ).toMatchSnapshot('cache: true');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
        },
      },
      schema,
    ),
  ).toMatchSnapshot('cache: false');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: '123',
        },
      },
      schema,
    ),
  ).toMatchSnapshot('actions: string');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: ['123'],
        },
      },
      schema,
    ),
  ).toMatchSnapshot('actions: string[1]');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: ['123', '456'],
        },
      },
      schema,
    ),
  ).toMatchSnapshot('actions: string[2]');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: ['123', '456'],
          format: 'json',
        },
      },
      schema,
    ),
  ).toMatchSnapshot('json');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: ['123', '456'],
          format: 'raw',
        },
      },
      schema,
    ),
  ).toMatchSnapshot('raw');

  expect(
    Joi.validate(
      {
        query: {
          url: 'http://url.com',
          selector: 'selector',
          cache: false,
          actions: ['123', '456'],
          format: 'not support',
        },
      },
      schema,
    ),
  ).toMatchSnapshot('not support');
});
