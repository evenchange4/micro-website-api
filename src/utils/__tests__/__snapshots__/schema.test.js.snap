// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions: string 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": "123",
      "cache": false,
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`actions: string[1] 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": Array [
        "123",
      ],
      "cache": false,
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`actions: string[2] 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": Array [
        "123",
        "456",
      ],
      "cache": false,
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`cache: false 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "cache": false,
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`cache: true 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "cache": true,
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`invalid uri format 1`] = `
Object {
  "catch": [Function],
  "error": [ValidationError: child "query" fails because [child "url" fails because ["url" must be a valid uri]]],
  "then": [Function],
  "value": Object {
    "query": Object {
      "selector": "selector",
      "url": "123",
    },
  },
}
`;

exports[`json 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": Array [
        "123",
        "456",
      ],
      "cache": false,
      "format": "json",
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`not support 1`] = `
Object {
  "catch": [Function],
  "error": [ValidationError: child "query" fails because [child "format" fails because ["format" must be one of [raw, json]]]],
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": Array [
        "123",
        "456",
      ],
      "cache": false,
      "format": "not support",
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`raw 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "actions": Array [
        "123",
        "456",
      ],
      "cache": false,
      "format": "raw",
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;

exports[`required 1`] = `
Object {
  "catch": [Function],
  "error": [ValidationError: child "query" fails because [child "url" fails because ["url" is required]]],
  "then": [Function],
  "value": Object {
    "query": Object {},
  },
}
`;

exports[`should return result 1`] = `
Object {
  "catch": [Function],
  "error": null,
  "then": [Function],
  "value": Object {
    "query": Object {
      "selector": "selector",
      "url": "http://url.com",
    },
  },
}
`;
